internal/resources/metal/organization_member/models.go

```diff
@@ -0,0 +1,76 @@
+package organizationmember
+
+import (
+	"context"
+	"fmt"
+	"path"
+
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+
+	"github.com/packethost/packngo"
+)
+
+type ResourceModel struct {
+	ID             types.String `tfsdk:"id"`
+	Invitee        types.String `tfsdk:"invitee"`
+	InvitedBy      types.String `tfsdk:"invited_by"`
+	OrganizationID types.String `tfsdk:"organization_id"`
+	ProjectsIDs    types.Set    `tfsdk:"projects_ids"`
+	Nonce          types.String `tfsdk:"nonce"`
+	Message        types.String `tfsdk:"message"`
+	Created        types.String `tfsdk:"created"`
+	Updated        types.String `tfsdk:"updated"`
+	Roles          types.Set    `tfsdk:"roles"`
+	State          types.String `tfsdk:"state"`
+}
+
+// func (m *ResourceModel) parse(ctx context.Context, invitee string, members []packngo.Member, invitations []packngo.Invitation) diag.Diagnostics {
+func (m *ResourceModel) parse(ctx context.Context, member *member) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	if member.isMember() {
+		projectsList, _ := types.SetValueFrom(ctx, types.StringType, member.Member.Projects)
+		m.ProjectsIDs = projectsList
+		m.State = types.StringValue("active")
+
+		rolesList, _ := types.SetValueFrom(ctx, types.StringType, member.Member.Roles)
+
+		m.Roles = rolesList
+		m.OrganizationID = types.StringValue(member.Member.Organization.URL)
+
+	} else if member.isInvitation() {
+
+		projectsList, _ := types.SetValueFrom(ctx, types.StringType, member.Invitation.Projects)
+		m.ProjectsIDs = projectsList
+
+		m.State = types.StringValue("invited")
+
+		rolesList, _ := types.SetValueFrom(ctx, types.StringType, member.Invitation.Roles)
+		m.Roles = rolesList
+
+		m.OrganizationID = types.StringValue(member.Invitation.Organization.Href)
+		m.Created = types.StringValue(member.Invitation.CreatedAt.String())
+		m.Updated = types.StringValue(member.Invitation.UpdatedAt.String())
+		m.Nonce = types.StringValue(member.Invitation.Nonce)
+
+		m.InvitedBy = types.StringValue(path.Base(member.Invitation.InvitedBy.Href))
+		m.ID = types.StringValue(member.Invitation.ID)
+	}
+	return diags
+}
+
+func FindMember(invitee string, members []packngo.Member, invitations []packngo.Invitation) (*member, error) {
```

Comment:
This should go in the same file where the `member` type is defined, and it should be private if it's only used by things in the `organization_member` package.